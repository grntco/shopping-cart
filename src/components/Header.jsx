import styled from 'styled-components'
import { Link } from 'react-router-dom'
import cartIcon from '../assets/icons/shopping-cart.svg'

const Nav = styled.nav`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 24px;
    border-bottom: 1px solid black;
`

const NavLinks = styled.ul`
    display: flex;
    align-items: center;
    gap: 24px;
    list-style: none;
`

const NavLink = styled(Link)`
    text-decoration: none;
    color: black;
    font-size: 1.2rem;

    &:hover {
        opacity: 0.6;
    }
`
const NavLogo = styled(NavLink)`
    display: flex;
    align-items: center;
    gap: 4px;
    font-size: 2rem;
`

const Header = () => {
    return (
        <header data-testid='header'>
            <Nav>
                <NavLogo to='/'>
                    <img src={cartIcon} alt='cart icon' />
                    <p>CartMart</p>
                </NavLogo>
                <NavLinks>
                    <li>
                        <NavLink to='/products'>Products</NavLink>
                    </li>
                    <li>
                        <NavLink to='/cart'>
                            <img src={cartIcon} alt='cart icon' />
                        </NavLink>
                    </li>
                </NavLinks>
            </Nav>
        </header>
    )
}

export default Header
