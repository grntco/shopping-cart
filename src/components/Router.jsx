import { createBrowserRouter, RouterProvider } from 'react-router-dom'
import App from './App'
import NotFound from './NotFound'
// import CartPage from './pages/cart/CartPage'

const Router = () => {
    const router = createBrowserRouter([
        {
            path: '/',
            element: <App />,
            errorElement: <NotFound />,
        },
        {
            path: '/:name',
            element: <App />,
        },
        {
            path: '/products/:id',
            element: <App />,
        },
    ])

    return <RouterProvider router={router} />
}

export default Router
