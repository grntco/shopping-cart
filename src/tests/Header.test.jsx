import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import Header from '../components/Header'
import Router from '../components/Router'

describe('Header renders', () => {
    render(
        <Router>
            <Header />
        </Router>,
    )
    it('renders header element', () => {
        expect(screen.getByTestId('header')).toMatchSnapshot()
    })
})
