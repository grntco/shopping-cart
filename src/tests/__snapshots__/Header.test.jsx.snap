// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header renders > renders header element 1`] = `
<header
  data-testid="header"
>
  <nav
    class="sc-beySPh ejyIdm"
  >
    <a
      class="sc-dmyCSP sc-hLQSwg fgQcUC kROMjo"
      href="/"
    >
      <img
        alt="cart icon"
        src="/src/assets/icons/shopping-cart.svg"
      />
      <p>
        CartMart
      </p>
    </a>
    <ul
      class="sc-guDLey FlvQP"
    >
      <li>
        <a
          class="sc-dmyCSP fgQcUC"
          href="/products"
        >
          Products
        </a>
      </li>
      <li>
        <a
          class="sc-dmyCSP fgQcUC"
          href="/cart"
        >
          <img
            alt="cart icon"
            src="/src/assets/icons/shopping-cart.svg"
          />
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
